arch/xtensa/include/xtensa-asm2-s.h

```diff
@@ -330,13 +337,19 @@ _xstack_returned_\@:
 	wsr.PS a0
 _not_l1:
 
-	/* Unmask EXCM bit so C code can spill/fill in window
-	 * exceptions.  Note interrupts are already fully masked by
-	 * INTLEVEL, so this is safe.
+	/* Setting up the cross stack call below has states where the
+	 * resulting frames are invalid/non-reentrant, so we can't
+	 * allow nested interrupts.  But we do need EXCM unmasked, as
+	 * we use CALL/ENTRY instructions in the process and need to
+	 * handle exceptions to spill caller/interruptee frames.  Use
+	 * PS.INTLEVEL at maximum to mask all interrupts and stash the
+	 * current value in our designated EPS register (which is
+	 * guaranteed unused across the call)
 	 */
-	rsr.PS a0
+	rsil a0, 0xf
```

Comment:
`XCHAL_NMILEVEL` ?

Reply:
In theory it shouldn't matter.  In practice NMILEVEL is set in a core-isa.h header and could be wrong if we mess up the toolchain setup.

FWIW, by my read the ISA spec is pretty clear that INTLEVEL can have any value from 0 to 15.  It's the inbound interrupts that might be implemented only at certain levels.